src/modules/did/did.service.ts

```diff
@@ -413,22 +415,42 @@ export class DIDService implements OnModuleInit {
 
         const token = await this.ipfsStore.get(serviceEndpoint);
 
-        const decodedData = jwt.decode(token) as {
-          claimData: Record<string, string>;
-        };
+        if (isJWT(token)) {
+          const decodedData = jwt.decode(token) as {
+            claimData: Record<string, string>;
+          };
 
-        if (!decodedData) {
-          return { serviceEndpoint, ...rest };
-        }
+          if (!decodedData) {
+            return { serviceEndpoint, ...rest };
+          }
+
+          const { claimData, ...claimRest } = decodedData;
 
-        const { claimData, ...claimRest } = decodedData;
+          return {
+            serviceEndpoint,
+            ...rest,
+            ...claimData,
+            ...claimRest,
+          };
+        }
 
-        return {
-          serviceEndpoint,
-          ...rest,
-          ...claimData,
-          ...claimRest,
-        };
+        try {
+          const data = JSON.parse(token);
+          if (isVerifiableCredential(data)) {
+            return {
+              serviceEndpoint,
+              ...rest,
+              verifiableCredentials: data,
+            };
+          }
+          return {
+            serviceEndpoint,
+            ...rest,
+            ...data,
+          };
+        } catch {
```

Comment:
Do you think it makes sense to log at `debug` level here? It could get pretty verbose if we have lots of these, but it could be good to have to troubleshoot